lib/src/rewrite.rs

```diff
@@ -145,6 +145,23 @@ impl<'repo> CommitRewriter<'repo> {
         self.new_parents = new_parents;
     }
 
+    /// Update the intended new parents by replacing any occurrence of
+    /// `old_parent` by `new_parents`
+    pub fn replace_parent(&mut self, old_parent: &CommitId, new_parents: &[CommitId]) {
+        if self.new_parents.contains(old_parent) {
+            self.new_parents = std::mem::take(&mut self.new_parents)
+                .into_iter()
+                .flat_map(|id| {
+                    if id == *old_parent {
+                        new_parents.to_vec()
+                    } else {
+                        vec![id]
+                    }
+                })
+                .collect();
```

Comment:
nit: perhaps, this can be rewritten as `.iter().position()` + `.splice()`, but we might have to ensure that `new_parents` have no dupes in `self.new_parents`